sys/include/usb/dfu.h

```diff
@@ -0,0 +1,168 @@
+    USBUS_DFU_STATE_DFU_IDLE,               /**< DFU runtime mode idle */
+    USBUS_DFU_STATE_DFU_DL_SYNC,            /**< DFU download synchronization */
+    USBUS_DFU_STATE_DFU_DL_BUSY,            /**< DFU download busy */
+    USBUS_DFU_STATE_DFU_DL_IDLE,            /**< DFU download idle */
+    USBUS_DFU_STATE_DFU_MANIFEST_SYNC,      /**< DFU manifest synchronization */
+    USBUS_DFU_STATE_DFU_MANIFEST,           /**< DFU manifest mode */
+    USBUS_DFU_STATE_DFU_MANIFEST_WAIT_RST,  /**< DFU manifest wait for CPU reset */
+    USBUS_DFU_STATE_DFU_UP_IDLE,            /**< DFU upload idle */
+    USBUS_DFU_STATE_DFU_ERROR               /**< DFU internal error */
+} usbus_dfu_state_t;
+
+/**
+ * @brief USB DFU interface descriptor
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t length;              /**< Descriptor length */
+    uint8_t type;                /**< Descriptor type */
+    uint8_t attribute;           /**< Descriptor attributes flags */
+    uint16_t detach_timeout;     /**< Descriptor detach timeout (ms) */
+    uint16_t xfer_size;          /**< Descriptor transaction size */
+    uint16_t bcd_dfu;            /**< Descriptor bcd version */
+} usb_desc_if_dfu_t;
+
+/**
+ * @brief USB DFU get_status control request packet
+ */
+typedef struct __attribute__((packed))  {
+    uint8_t status;             /**< DFU status response */
+    uint32_t timeout : 24;      /**< DFU timeout (ms) response */
+    uint8_t state;              /**< DFU internal state machine */
+    uint8_t string;             /**< DFU string */
+} dfu_get_status_pkt_t;
+
+/**
+ * @brief USBUS DFU device interface context
+ */
+typedef struct usbus_dfu_device {
+    usbus_handler_t handler_ctrl;           /**< Control interface handler */
+    usbus_interface_t iface;                /**< Control interface */
+    usbus_descr_gen_t dfu_descr;            /**< DFU descriptor generator */
+    usbus_string_t slot0_str;               /**< Descriptor string for Slot 0 */
+#ifdef MODULE_RIOTBOOT_USB_DFU
+    usbus_interface_alt_t iface_alt_slot1;  /**< Alt interface for secondary slot */
+    usbus_string_t slot1_str;               /**< Descriptor string for Slot 1 */
+    riotboot_flashwrite_t writer;           /**< DFU firmware update state structure */
+#endif
+    bool skip_signature;                    /**< Skip RIOTBOOT signature status */
+    usbus_t *usbus;                         /**< Ptr to the USBUS context */
+    unsigned mode;                          /**< 0 - APP mode, 1 DFU mode */
+    unsigned selected_slot;                 /**< Slot used for upgrade */
+    usbus_dfu_state_t dfu_state;            /**< Internal DFU state machine */
+} usbus_dfu_device_t;
+
+/**
+ * @brief DFU initialization function
+ *
+ * @param   usbus   USBUS thread to use
+ * @param   handler DFU device struct
+ * @param   mode    DFU start mode (0 runtime mode / 1 dfu mode)
+ */
+void usbus_dfu_init(usbus_t *usbus, usbus_dfu_device_t *handler, unsigned mode);
```

Comment:
Would it be possible to move these definitions to `sys/include/usb/usbus/dfu.h` and keep this file clean of implementation specifics and only on generic DFU macros and structs? :pleading_face: